etcd/src/main/java/org/jclouds/etcd/handlers/EtcdErrorHandler.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.etcd.handlers;
+
+import static org.jclouds.util.Closeables2.closeQuietly;
+
+import java.io.IOException;
+
+import javax.annotation.Resource;
+
+import org.jclouds.http.HttpCommand;
+import org.jclouds.http.HttpErrorHandler;
+import org.jclouds.http.HttpResponse;
+import org.jclouds.http.HttpResponseException;
+import org.jclouds.logging.Logger;
+import org.jclouds.rest.ResourceNotFoundException;
+import org.jclouds.util.Strings2;
+
+import org.jclouds.etcd.EtcdApiMetadata;
+import com.google.common.base.Throwables;
+
+/**
+ * Handle errors and propagate exception
+ */
+public class EtcdErrorHandler implements HttpErrorHandler {
+   @Resource
+   protected Logger logger = Logger.NULL;
+
+   public void handleError(HttpCommand command, HttpResponse response) {
+
+      String message = parseMessage(response);
+      Exception exception = null;
+      try {
+
+         message = message != null ? message
+               : String.format("%s -> %s", command.getCurrentRequest().getRequestLine(), response.getStatusLine());
+
+         switch (response.getStatusCode()) {
+            case 404:
+               if (command.getCurrentRequest().getMethod().equals("DELETE")) {
+                  if (command.getCurrentRequest().getEndpoint().getPath().startsWith(
+                        "/" + EtcdApiMetadata.API_VERSION + "/members") && message.contains("No such member")) {
+                     exception = new ResourceNotFoundException(message,
+                           new HttpResponseException(command, response, message));
+                  }
+               }
+               break;
+            case 409:
+               if (command.getCurrentRequest().getMethod().equals("POST")) {
+                  if (command.getCurrentRequest().getEndpoint().getPath().endsWith(
+                        EtcdApiMetadata.API_VERSION + "/members") && message.contains("etcdserver: ID exists"))
+                     exception = new HttpResponseException(message, command, response);
```

Comment:
I'd remove the method and path checks, and throw in general a `ResourceAlreadyExistsException ` exception if the ID exists, and the `HttpResponseException` otherwise.